src/libs/portal.js

```diff
@@ -242,7 +297,7 @@ function getCal(db, user, callback) {
 			const data = ical.parseICS(body);
 
 			// School Portal does not give a 404 if calendar is invalid. Instead, it gives an empty calendar.
-			// Unlike Canvas, the Portal is guaranteed to contain some sort of data within a span of a year.
```

Comment:
This wasn't a typo 😉 

Canvas so far has been reliable. If there's no events, it may mean it's the summer and canvas classes haven't been registers. Portal on the other hand, should theoretically return something any time during the year because the website says it will return everything "for the past 2 months and the upcoming 12 months." If the feed is empty, then something's wrong and therefore we should error just like we are now.